src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -3600,12 +3599,11 @@ impl Step for TestFloatParse {
             &[],
         );
 
-        cargo_run.arg("--");
-        if builder.config.args().is_empty() {
-            // By default, exclude tests that take longer than ~1m.
-            cargo_run.arg("--skip-huge");
-        } else {
-            cargo_run.args(builder.config.args());
+        if env::var("FLOAT_PARSE_TESTS_SKIP_HUGE").unwrap_or_default() == "1" {
+            cargo_run
+                .arg("--")
+                // By default, exclude tests that take longer than ~1m.
+                .arg("--skip-huge");
```

Comment:
I think this might be missing another branch here - with the change change it looks like the tests don't run at all unless `FLOAT_PARSE_TESTS_SKIP_HUGE=1`.

The logic should be: always run by default with `--skip-huge`. If the user somehow indicates that slow tests should _not_ be skipped, don't pass `--skip-huge`. (So if this is done via env it should be something like `FLOAT_PARSE_TESTS_NO_SKIP_HUGE`).

Reply:
I think it's quite annoying not to run all tests by default and to have to pass something like `NO_SKIP_HUGE`. If we don't want to run them by default, why the tool itself doesn't do it by default instead of requiring bootstrap to pass `--skip-huge`? Anyway, not a big deal — I can do that.

Reply:
The reasoning is that when running the whole testsuite or in CI, about a minute of tests here is good enough. However, if it is run on its own, you probably don't want to forget to run the more thorough tests because of not passing a flag. I don't know what is best here, maybe the test should skip huge if RUSTC_BOOTSTRAP is set? (If you want to do that here it's easy enough, just set [`cfg.skip_huge`](https://github.com/rust-lang/rust/blob/dff3e7ccd4a18958c938136c4ccdc853fcc86194/src/etc/test-float-parse/src/main.rs#L72-L73) if the env is set).

Anyway the change looks good as-is 👍